qiskit/dagcircuit/dagcircuit.py

```diff
@@ -1898,6 +1898,64 @@ def count_ops_longest_path(self):
                 op_dict[name] += 1
         return op_dict
 
+    def _get_qubit_input_output_node(self, qubit, in_or_out=False):
+        """Returns qubit and input or output node from an index."""
+        nodes = self.output_map if in_or_out else self.input_map
+        # Check if index is passed
+        if isinstance(qubit, int):
+            if qubit >= self.num_qubits():
+                raise DAGCircuitError(f"Qubit index {qubit} is out of range")
+            qubit = list(nodes.keys())[qubit]
+        if qubit not in self.qubits:
+            raise DAGCircuitError("Qubit was not found in circuit")
+        return (qubit, nodes.get(qubit, None))
+
+    def get_qubit_input_node(self, qubit):
+        """Returns qubit and input node from a qubit index."""
+        return self._get_qubit_input_output_node(qubit)
+
+    def get_qubit_output_node(self, qubit):
+        """Returns qubit and output node from a qubit index."""
+        return self._get_qubit_input_output_node(qubit, True)
+
+    def get_causal_cone(self, qubit_index):
+        """Returns causal cone of a qubit."""
+        # Check if the qubit index is in range, else throw an error.
+        if qubit_index >= self.num_qubits():
+            raise DAGCircuitError(f"Qubit index {qubit_index} is out of range")
+
+        # Retrieve the output node and the qubit
+        qubit, output_node = self.get_qubit_output_node(qubit_index)
```

Comment:
If you take a qubit object you can do something like:



```suggestion
        output_node = self.output_map.get(qubit, None)
        if output_node is None:
            raise DAGCircuitError(f"Qubit {qubit} is not in this DAGCircuit")
```

Reply:
As per your comments below, this function will be promptly removed.